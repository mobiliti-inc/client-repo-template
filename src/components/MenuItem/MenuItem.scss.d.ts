// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'fades-in-out': string;
  'menu-item': string;
  'menu-item--active': string;
  'menu-item__left': string;
  'menu-item__left__text': string;
  'menu-item__right': string;
  'nested': string;
}
declare var cssExports: CssExports;
export = cssExports;
