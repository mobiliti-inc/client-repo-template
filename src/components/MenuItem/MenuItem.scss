@import '../../styles/variables';

.menu-item {
    display: grid;
    grid-template-columns: auto auto;
    width: 100%;
    cursor: pointer;
    position: relative;
    color: inherit;
    &:hover {
        background-color:rgba(0, 0, 0, 0.1);
    }

    & > span {
        background-color:rgba(0, 0, 0, 0.24);
        display: none;
        position: absolute;
        width: 100%;
        height: 100%;
        left: 0;
        top: 0;
    }

    &--active > .nested {
        background-color:rgba(0, 0, 0, 0.42);
    }

    &--active > span {
        display: block;
    }

    &__left {
        display: grid;
        grid-template-columns: min-content min-content;
        grid-gap: 1em;
        align-items: center;
        & > img {
            max-width: 24px;
        }
        &__text {
            display: flex;
            font-family: $futura;
            font-style: normal;
            font-weight: 500;
            font-size: 14px;
            line-height: 19px;
            align-items: center;
        }
    }

    &__right {
        justify-self: flex-end;
        display: flex;
        align-items: center;
        justify-content: flex-end;
        width: 12px;
    }
}


.nested {
    display: block;
}