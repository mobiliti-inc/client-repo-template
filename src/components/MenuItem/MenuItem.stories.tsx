import { storiesOf } from "@storybook/react";
import { boolean, object } from "@storybook/addon-knobs";
import React, { useState} from "react";
import { MenuItem } from "../";
import { dashboard, inventory } from '../../assets';

const  bottomContent = () => {
	const [isActive, setIsActive] = useState(false);
	return (<MenuItem style={{'padding': '0 35px'}} isChild isActive={isActive} onItemClicked={() => setIsActive( prev => !prev)} isIconUp={isActive} menuItemIcon={inventory} menuItemText="Inventory" />);
};

const DefaultMenuItem = () => {
	const [isActive, setIsActive] = useState(false);
	return (
		<MenuItem
			showRightIcon={boolean("showRightIcon", true)}
			isActive={boolean("isActive", isActive)}
			onItemClicked={() => setIsActive( prev => !prev)}
			isIconUp={isActive} menuItemIcon={dashboard}
			menuItemText="Dashboard"
			style={ object('style', { background: '#4996ae' })}
			isChild={boolean("isChild", false)}
		/>);
};

storiesOf("MenuItem", module)
	.add("Default", () => <DefaultMenuItem />);
