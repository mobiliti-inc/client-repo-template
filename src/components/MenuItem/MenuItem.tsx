import React, { FC, ReactNode } from "react";
import classNames from 'classnames';

import "./MenuItem.scss";
import { chevronDown, chevronUp } from '../../assets';

type customStyle = {
	background?: string;
	color?: string;
	height?: string;
	width?: string;
	padding?: string;
};

interface IProps {
	style?: customStyle;
	menuItemIcon?: string;
	menuItemText: ReactNode;
	showRightIcon?: boolean;
	isIconUp?: boolean;
	onItemClicked: () => void;
	isActive?: boolean;
	isChild?: boolean;
}

const MenuItem: FC<IProps> = ({ style, menuItemIcon, menuItemText, showRightIcon, isIconUp, onItemClicked, isActive, isChild }) => (
		<div
		style={style}
		styleName={classNames("menu-item", { "menu-item--active": isActive})}
		onClick={onItemClicked}
		role="presentation"
		>
			<div styleName="menu-item__left">
				<img src={menuItemIcon} alt="Menu Item Icon" />
				<span styleName="menu-item__left__text">{menuItemText}</span>
			</div>
			{
				showRightIcon &&
				<div styleName="menu-item__right">
					{isIconUp && <img className="upIcon" src={chevronUp} alt="Icon up" />}
					{!isIconUp && <img className="downIcon" src={chevronDown} alt="Icon down" />}
				</div>
			}
			<span styleName={classNames({nested: isChild})}/>
		</div>
	);

export default MenuItem;
