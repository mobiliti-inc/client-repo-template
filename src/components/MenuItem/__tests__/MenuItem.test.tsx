import React from 'react';

import MenuItem from '../MenuItem';

describe('MenuItem', () => {
	const defaultProps = {
		style: {},
		menuItemIcon: 'chevron.png',
		menuItemText: 'DashBoard',
		showRightIcon: false,
		isIconUp: false,
		onItemClicked: jest.fn(),
		isActive: false,
		isChild: false
	};
	it('renders', () => {
		const subject = shallow(<MenuItem  {...defaultProps} />);
		expect(subject).toMatchSnapshot();
	});

	it('renders with right icon', () => {
		const newProps = {...defaultProps, showRightIcon: true};
		const subject = shallow(<MenuItem  {...newProps}/>);
		expect(subject).toMatchSnapshot();
		expect(subject.find('.menu-item__right')).toHaveLength(1);
	});

	it('renders with up icon shown', () => {
		const newProps = {...defaultProps, showRightIcon: true, isIconUp: true};
		const subject = shallow(<MenuItem  {...newProps}/>);
		expect(subject).toMatchSnapshot();
		expect(subject.find('.upIcon')).toHaveLength(1);
	});
});
