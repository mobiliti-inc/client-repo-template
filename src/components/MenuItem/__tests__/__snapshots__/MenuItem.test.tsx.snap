// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem renders 1`] = `
<div
  className="menu-item"
  onClick={[MockFunction]}
  role="presentation"
  style={Object {}}
>
  <div
    className="menu-item__left"
  >
    <img
      alt="Menu Item Icon"
      src="chevron.png"
    />
    <span
      className="menu-item__left__text"
    >
      DashBoard
    </span>
  </div>
  <span
    className=""
  />
</div>
`;

exports[`MenuItem renders with right icon 1`] = `
<div
  className="menu-item"
  onClick={[MockFunction]}
  role="presentation"
  style={Object {}}
>
  <div
    className="menu-item__left"
  >
    <img
      alt="Menu Item Icon"
      src="chevron.png"
    />
    <span
      className="menu-item__left__text"
    >
      DashBoard
    </span>
  </div>
  <div
    className="menu-item__right"
  >
    <img
      alt="Icon down"
      className="downIcon"
      src="test-file-stub"
    />
  </div>
  <span
    className=""
  />
</div>
`;

exports[`MenuItem renders with up icon shown 1`] = `
<div
  className="menu-item"
  onClick={[MockFunction]}
  role="presentation"
  style={Object {}}
>
  <div
    className="menu-item__left"
  >
    <img
      alt="Menu Item Icon"
      src="chevron.png"
    />
    <span
      className="menu-item__left__text"
    >
      DashBoard
    </span>
  </div>
  <div
    className="menu-item__right"
  >
    <img
      alt="Icon up"
      className="upIcon"
      src="test-file-stub"
    />
  </div>
  <span
    className=""
  />
</div>
`;
